### PR TITLE
Extend image proxy to work with cover art

### LINKED ISSUE #76: Port thumbnail server to work with images associated with audio
The goal is to update the thumbnail server `willnorris/imageproxy` to work with images that are associated with audio files, which would typically include album art, front or back covers, artist images etc.

Branched from WordPress/openverse-frontend#61.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Closes #76

## Description
This PR adds endpoints to generate thumbnails for album art associated with audio sets

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — CHANGES_REQUESTED** [review]
I got it to work by manually adding an audioset, adding the `audio_set_id` to the audio row, and visiting the thumb url.

`INSERT INTO "public"."api_audioset" ("id", "created_on", "updated_on", "identifier", "foreign_identifier", "title", "foreign_landing_url", "creator", "creator_url", "url", "filesize") VALUES
(4, '2021-08-13 03:21:57.193204+00', '2021-08-13 03:21:57.193204+00', '40e6215d-b5c6-4896-987c-f30f3678f608', '31', 'Pouce !', 'https://www.jamendo.com/album/31/pouce', 'fake', 'https://fake.com', 'https://usercontent.jamendo.com?type=album&id=31&width=200&trackid=217', 200);`

Then added the resulting `id` as the `audio_set_id` for audio where identifier is `28c40835-f14d-4b81-8847-bbb091a64f35`.

`http://localhost:8000/v1/audio/28c40835-f14d-4b81-8847-bbb091a64f35/thumb`

CleanShot 2021-08-12 at 23 24 26@2x

My only real feedback is that I don't love the `audio_set` column names `url` for the thumbnail url. 

I'm also wondering if it makes sense to store the thumbnail url on the individual audio rows as a backup if anything happens to the audio set, or maybe we just need some other protection to make sure audio sets aren't deleted/lost/disconnected in the future.

**Review by zackkrida — APPROVED** [review]

File: openverse-api/catalog/api/views/audio_views.py [hunk 1]
```diff
@@ -237,3 +239,27 @@ class AudioStats(MediaStats):
...
+    """
+
+    queryset = Audio.objects.all()
+
+    def get(self, request, identifier, format=None):
+        serialized = ProxiedImageSerializer(data=request.data)
+        serialized.is_valid()
+        try:
+            audio = Audio.objects.get(identifier=identifier)
+            image_url = audio.audio_set.url
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply openverse-api/catalog/api/views/audio_views.py:256]: Right, thanks for pointing it out. I should probably add handling for this before merging.

[the remark this answers is not in the document]
Comment (dhruvkb) [reply openverse-api/catalog/api/views/audio_views.py:256]: Added handling for missing audio, audio set and cover art in e1b4e2a.

[the remark this answers is not in the document]
Comment (zackkrida) [reply openverse-api/catalog/api/views/audio_views.py:256]: Perfect, thanks Dhruv.

**Review by obulat — APPROVED** [review]
Thumbnail images work great.
Thank you for adding changes so quickly!

I was wondering how cache works with thumbnails. It seems that many audio files will have the same thumbnail (since the image is the same for all audios in the audio_set, at least on jamendo).

**Comment by dhruvkb** [comment]
@obulat the image proxy we use for scaling has [caching functionality](https://github.com/willnorris/imageproxy#cache) built-in, although we have not configured it in the dev environment. The documentation covers most caching setups so it should not pose significant difficulty.

**Pull request merged**